### PR TITLE
Inject CA trust bundle into managed containers and set SSL_CERT_DIR

### PR DESCRIPTION
**Pull request by osmman**
related to https://issues.redhat.com/browse/SECURESIGN-703 and https://issues.redhat.com/browse/SECURESIGN-1037

### CODE REVIEW

File: internal/controller/annotations/annotations.go [hunk 1]
```diff
@@ -6,4 +6,23 @@ const (
...
+
+	// TrustedCA Annotation to specify name of ConfigMap with additional bundle of trusted CA
+	TrustedCA = "rhtas.redhat.com/trusted-ca"
 )
+
+var inheritable = []string{
+	TrustedCA,
+}
+
+func FilterInheritable(annotations map[string]string) map[string]string {
```
Comment (bouskaJ) [inline internal/controller/annotations/annotations.go:18]: What is purpose of this function? Why you just don't use `annotations[TrustedCA]`?

**Comment by bouskaJ** [comment]
Can you create `e2e` test for this feature?

**Comment by bouskaJ** [comment]
> Can you create `e2e` test for this feature?

I mean can modify the CI to use keycloak (self signed) route instead of the insecure service https://github.com/securesign/secure-sign-operator/blob/main/.github/workflows/main.yml#L187 This should verify `kuberoot` CA injection AFIK

**Comment by osmman** [comment]
> > Can you create `e2e` test for this feature?
> 
> I mean can modify the CI to use keycloak (self signed) route instead of the insecure service https://github.com/securesign/secure-sign-operator/blob/main/.github/workflows/main.yml#L187 This should verify `kuberoot` CA injection AFIK
> 
> I am not sure how to test the annotation injection. But with that ^^ we will know that the volume is correctly mounted and the `ENV` property works fine

Yes, that will be good to modify Keycloack deployment on Kind cluster. I think Openshift CI uses self signed certificates if I remember correctly.

**Pull request merged**